feat: add loading animations to sign-in and sign-up buttons

Ethan225300 opened the pull request:
- Implemented loading animations for sign-in and sign-up buttons during API fetch and mutation.
- Change the highet of 4th card on dashboard from 140px to 139px.

Resolve CP-129

<img width="290" alt="image" src="https://github.com/user-attachments/assets/de7b590c-d875-439b-be07-593a950692b4">

<!-- link to tickets -->

---

## What happens here?

## How do I know this is working?

## Any notes?

<!--
  New PR Checklist

- Have you tag the right reviewer?
- Have you dm the reviewer?
- Maybe @ the reviewer in channel just to be safe

- is CI passing in your pr?
- Have you update the ticket with the right status?
-->

Chris-Wang commented on apps/client/src/module/auth/login-form.tsx:
```diff
@@ -16,29 +16,31 @@ import { AuthFormContainer } from "./auth-form-container";
+  const [loading, setLoading] = useState(false); 
 
   type LoginForm = z.infer<typeof formSchema>;
-  // source of truth for this login form <= this is the law, is the king, is the absolute authority for this form - K总
```
these changes should not be here

Chris-Wang commented on apps/client/src/module/auth/login-form.tsx:
```diff
@@ -16,29 +16,31 @@ import { AuthFormContainer } from "./auth-form-container";
-  // source of truth for this login form <= this is the law, is the king, is the absolute authority for this form - K总
 
   const formSchema = z.object({
     email: z
```
these changes should not be here

Chris-Wang commented on apps/client/src/module/auth/login-form.tsx:
```diff
@@ -16,29 +16,31 @@ import { AuthFormContainer } from "./auth-form-container";
     remember: z.string(),
   });
 
-  // Need this to enforce default value to match the LoginForm schema, because if you do it in the `useForm` hook, it turns it into a `Partial<LoginForm>` type.
```
these changes should not be here

Chris-Wang commented on apps/client/src/module/auth/login-form.tsx:
```diff
@@ -16,29 +16,31 @@ import { AuthFormContainer } from "./auth-form-container";
     email: "",
     password: "",
     remember: "",
-  } satisfies LoginForm; // Satifies make sure this object you are making can satisfy the LoginForm schema.
```
these changes should not be here

Chris-Wang commented on apps/client/src/module/auth/login-form.tsx:
```diff
@@ -65,56 +74,77 @@ export function LoginForm() {
+            className="inline-block h-16 w-16 animate-spin rounded-full border-8 border-solid border-blue-500 border-t-transparent"
+            role="status"
+          >
+            <span className="sr-only">Loading...</span>
```
why we need to have this span?

Chris-Wang reviewed (changes requested):
After login, I get weird shadow on left bar:
<img width="352" alt="image" src="https://github.com/user-attachments/assets/04919518-0e1a-4da2-a2a2-ddae2dc471f2">
<img width="478" alt="image" src="https://github.com/user-attachments/assets/6dc6d175-4bef-4901-9385-bc7da5d10adb">

Chris-Wang reviewed (changes requested):
Still get shadow issue
and you need to rewrite commit msg

Commit:
feat: add loading animations to sign-in and sign-up buttons
Implemented loading animations for sign-in and sign-up buttons during API fetch and mutation.
Change the highet of 4th card on dashboard from 140px to 139px.
Resolve CP-129

Chris-Wang reviewed (approved):
LGTM
